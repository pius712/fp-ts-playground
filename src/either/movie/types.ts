export type MovieCategory = 'BOXOFFICE' | 'COMMING'
export type  MovieResponse = MovieCategoryList<'BOXOFFICE'> | MovieCategoryList<'COMMING'>
export type MovieCategoryList<T extends MovieCategory> = {
    "movieChartList": {
        'BOXOFFICE': Movie[]
    }
}

export interface Movie {
    "movieCode": number;// 74977,
    "movieTitle": string;// "아바타: 물의 길",
    "posterImageUrl": string;// "/20221215_185/1671091761840XXpCR_JPEG/movie_image.jpg",
    "weekendAudience": number;// 1274184,
    "openDate": null,
    // "pointSum": 0,
    // "joinNumber": 0,
    "reserveRatio": null,
    "saleRate": null,
    "salePossibleYn": boolean,
    "lendingPossibleYn": boolean,
    "adult":boolean;// false,
    // "korGradeList": [{ "code": "1001002", "name": "12세 관람가" }],
    // "lastKoreanGrade": { "code": "1001002", "name": "12세 관람가" },
    // "usGradeList": [],
    "lastUsGrade": null,
    // "theater": 0,
    "point": string; // "0.00",
    "formattedWeekendAudience": string; // "1,274,184",
    "formattedReserveRatio": string; // "0.00",
    "formattedSaleRate": string; // "0.00"
    [key:string]: any
}