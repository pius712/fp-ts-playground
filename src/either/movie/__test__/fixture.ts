import { MovieResponse } from "../types"

export const boxOffice = {
    "movieChartList":
        {
            "BOXOFFICE":
                [{
                    "movieCode": 74977,
                    "movieTitle": "아바타: 물의 길",
                    "posterImageUrl": "/20221215_185/1671091761840XXpCR_JPEG/movie_image.jpg",
                    "weekendAudience": 1274184,
                    "openDate": null,
                    "pointSum": 0,
                    "joinNumber": 0,
                    "reserveRatio": null,
                    "saleRate": null,
                    "salePossibleYn": false,
                    "lendingPossibleYn": false,
                    "adult": false,
                    "korGradeList": [{"code": "1001002", "name": "12세 관람가"}],
                    "lastKoreanGrade": {"code": "1001002", "name": "12세 관람가"},
                    "usGradeList": [],
                    "lastUsGrade": null,
                    "theater": 0,
                    "point": "0.00",
                    "formattedWeekendAudience": "1,274,184",
                    "formattedReserveRatio": "0.00",
                    "formattedSaleRate": "0.00"
                }, {
                    "movieCode": 184509,
                    "movieTitle": "영웅",
                    "posterImageUrl": "/20221213_158/1670910727328mpqYu_JPEG/movie_image.jpg",
                    "weekendAudience": 518865,
                    "openDate": null,
                    "pointSum": 0,
                    "joinNumber": 0,
                    "reserveRatio": null,
                    "saleRate": null,
                    "salePossibleYn": false,
                    "lendingPossibleYn": false,
                    "adult": false,
                    "korGradeList": [{"code": "1001002", "name": "12세 관람가"}],
                    "lastKoreanGrade": {"code": "1001002", "name": "12세 관람가"},
                    "usGradeList": [],
                    "lastUsGrade": null,
                    "theater": 0,
                    "point": "0.00",
                    "formattedWeekendAudience": "518,865",
                    "formattedReserveRatio": "0.00",
                    "formattedSaleRate": "0.00"
                }, {
                    "movieCode": 208508,
                    "movieTitle": "젠틀맨",
                    "posterImageUrl": "/20221228_75/1672203377094bQgf9_JPEG/movie_image.jpg",
                    "weekendAudience": 99110,
                    "openDate": null,
                    "pointSum": 0,
                    "joinNumber": 0,
                    "reserveRatio": null,
                    "saleRate": null,
                    "salePossibleYn": false,
                    "lendingPossibleYn": false,
                    "adult": false,
                    "korGradeList": [{"code": "1001003", "name": "15세 관람가"}],
                    "lastKoreanGrade": {"code": "1001003", "name": "15세 관람가"},
                    "usGradeList": [],
                    "lastUsGrade": null,
                    "theater": 0,
                    "point": "0.00",
                    "formattedWeekendAudience": "99,110",
                    "formattedReserveRatio": "0.00",
                    "formattedSaleRate": "0.00"
                }, {
                    "movieCode": 220733,
                    "movieTitle": "오늘 밤, 세계에서 이 사랑이 사라진다 해도",
                    "posterImageUrl": "/20221104_258/1667527490266scSO3_JPEG/movie_image.jpg",
                    "weekendAudience": 91483,
                    "openDate": null,
                    "pointSum": 0,
                    "joinNumber": 0,
                    "reserveRatio": null,
                    "saleRate": null,
                    "salePossibleYn": false,
                    "lendingPossibleYn": false,
                    "adult": false,
                    "korGradeList": [{"code": "1001002", "name": "12세 관람가"}],
                    "lastKoreanGrade": {"code": "1001002", "name": "12세 관람가"},
                    "usGradeList": [],
                    "lastUsGrade": null,
                    "theater": 0,
                    "point": "0.00",
                    "formattedWeekendAudience": "91,483",
                    "formattedReserveRatio": "0.00",
                    "formattedSaleRate": "0.00"
                }, {
                    "movieCode": 222301,
                    "movieTitle": "올빼미",
                    "posterImageUrl": "/20221123_65/1669168819647KdX9K_JPEG/movie_image.jpg",
                    "weekendAudience": 77160,
                    "openDate": null,
                    "pointSum": 0,
                    "joinNumber": 0,
                    "reserveRatio": null,
                    "saleRate": null,
                    "salePossibleYn": false,
                    "lendingPossibleYn": false,
                    "adult": false,
                    "korGradeList": [{"code": "1001003", "name": "15세 관람가"}],
                    "lastKoreanGrade": {"code": "1001003", "name": "15세 관람가"},
                    "usGradeList": [],
                    "lastUsGrade": null,
                    "theater": 0,
                    "point": "0.00",
                    "formattedWeekendAudience": "77,160",
                    "formattedReserveRatio": "0.00",
                    "formattedSaleRate": "0.00"
                }, {
                    "movieCode": 223799,
                    "movieTitle": "신비아파트 극장판 차원도깨비와 7개의 세계",
                    "posterImageUrl": "/20221212_211/1670833558390SLzsN_JPEG/movie_image.jpg",
                    "weekendAudience": 47672,
                    "openDate": null,
                    "pointSum": 0,
                    "joinNumber": 0,
                    "reserveRatio": null,
                    "saleRate": null,
                    "salePossibleYn": false,
                    "lendingPossibleYn": false,
                    "adult": false,
                    "korGradeList": [{"code": "1001001", "name": "전체 관람가"}],
                    "lastKoreanGrade": {"code": "1001001", "name": "전체 관람가"},
                    "usGradeList": [],
                    "lastUsGrade": null,
                    "theater": 0,
                    "point": "0.00",
                    "formattedWeekendAudience": "47,672",
                    "formattedReserveRatio": "0.00",
                    "formattedSaleRate": "0.00"
                }, {
                    "movieCode": 224788,
                    "movieTitle": "눈의 여왕5: 스노우 프린세스와 미러랜드의 비밀",
                    "posterImageUrl": "/20221221_206/16715988070200qULF_JPEG/movie_image.jpg",
                    "weekendAudience": 22991,
                    "openDate": null,
                    "pointSum": 0,
                    "joinNumber": 0,
                    "reserveRatio": null,
                    "saleRate": null,
                    "salePossibleYn": false,
                    "lendingPossibleYn": false,
                    "adult": false,
                    "korGradeList": [{"code": "1001001", "name": "전체 관람가"}],
                    "lastKoreanGrade": {"code": "1001001", "name": "전체 관람가"},
                    "usGradeList": [],
                    "lastUsGrade": null,
                    "theater": 0,
                    "point": "0.00",
                    "formattedWeekendAudience": "22,991",
                    "formattedReserveRatio": "0.00",
                    "formattedSaleRate": "0.00"
                }, {
                    "movieCode": 201642,
                    "movieTitle": "스위치",
                    "posterImageUrl": "/20230104_71/1672799791128mA8Km_JPEG/movie_image.jpg",
                    "weekendAudience": 19315,
                    "openDate": null,
                    "pointSum": 0,
                    "joinNumber": 0,
                    "reserveRatio": null,
                    "saleRate": null,
                    "salePossibleYn": false,
                    "lendingPossibleYn": false,
                    "adult": false,
                    "korGradeList": [{"code": "1001002", "name": "12세 관람가"}],
                    "lastKoreanGrade": {"code": "1001002", "name": "12세 관람가"},
                    "usGradeList": [],
                    "lastUsGrade": null,
                    "theater": 0,
                    "point": "0.00",
                    "formattedWeekendAudience": "19,315",
                    "formattedReserveRatio": "0.00",
                    "formattedSaleRate": "0.00"
                }, {
                    "movieCode": 222337,
                    "movieTitle": "몬스터 신부: 101번째 프로포즈",
                    "posterImageUrl": "/20221208_79/1670484193384Wix0b_JPEG/movie_image.jpg",
                    "weekendAudience": 15362,
                    "openDate": null,
                    "pointSum": 0,
                    "joinNumber": 0,
                    "reserveRatio": null,
                    "saleRate": null,
                    "salePossibleYn": false,
                    "lendingPossibleYn": false,
                    "adult": false,
                    "korGradeList": [{"code": "1001001", "name": "전체 관람가"}],
                    "lastKoreanGrade": {"code": "1001001", "name": "전체 관람가"},
                    "usGradeList": [],
                    "lastUsGrade": null,
                    "theater": 0,
                    "point": "0.00",
                    "formattedWeekendAudience": "15,362",
                    "formattedReserveRatio": "0.00",
                    "formattedSaleRate": "0.00"
                }, {
                    "movieCode": 222543,
                    "movieTitle": "극장판 뽀로로와 친구들: 바이러스를 없애줘!",
                    "posterImageUrl": "/20221108_225/1667872338018ez9cs_JPEG/movie_image.jpg",
                    "weekendAudience": 11967,
                    "openDate": null,
                    "pointSum": 0,
                    "joinNumber": 0,
                    "reserveRatio": null,
                    "saleRate": null,
                    "salePossibleYn": false,
                    "lendingPossibleYn": false,
                    "adult": false,
                    // "korGradeList": [{"code": "1001001", "name": "전체 관람가"}],
                    // "lastKoreanGrade": {"code": "1001001", "name": "전체 관람가"},
                    "usGradeList": [],
                    "lastUsGrade": null,
                    "theater": 0,
                    "point": "0.00",
                    "formattedWeekendAudience": "11,967",
                    "formattedReserveRatio": "0.00",
                    "formattedSaleRate": "0.00"
                }]
        }
} as any;

export const comming = {
    "movieChartList": {
        "COMMING": [{
            "movieCode": 184513,
            "movieTitle": "교섭",
            "posterImageUrl": "/20221227_112/16721270739480lDiS_JPEG/movie_image.jpg",
            "weekendAudience": 0,
            "openDate": "1.18",
            "pointSum": 10,
            "joinNumber": 1,
            "reserveRatio": null,
            "saleRate": null,
            "salePossibleYn": false,
            "lendingPossibleYn": false,
            "adult": false,
            "korGradeList": [{"code": "1001002", "name": "12세 관람가"}],
            "lastKoreanGrade": {"code": "1001002", "name": "12세 관람가"},
            "usGradeList": [],
            "lastUsGrade": null,
            "theater": 392,
            "formattedWeekendAudience": "0",
            "point": "10.00",
            "formattedReserveRatio": "0.00",
            "formattedSaleRate": "0.00"
        }, {
            "movieCode": 200745,
            "movieTitle": "유령",
            "posterImageUrl": "/20221228_48/1672193628576Mv91Y_JPEG/movie_image.jpg",
            "weekendAudience": 0,
            "openDate": "1.18",
            "pointSum": 0,
            "joinNumber": 0,
            "reserveRatio": null,
            "saleRate": null,
            "salePossibleYn": false,
            "lendingPossibleYn": false,
            "adult": false,
            "korGradeList": [{"code": "1001003", "name": "15세 관람가"}],
            "lastKoreanGrade": {"code": "1001003", "name": "15세 관람가"},
            "usGradeList": [],
            "lastUsGrade": null,
            "theater": 373,
            "formattedWeekendAudience": "0",
            "point": "0.00",
            "formattedReserveRatio": "0.00",
            "formattedSaleRate": "0.00"
        }, {
            "movieCode": 222535,
            "movieTitle": "프린스 챠밍",
            "posterImageUrl": "/20221227_39/1672132806007stVFH_JPEG/movie_image.jpg",
            "weekendAudience": 0,
            "openDate": "1.11",
            "pointSum": 0,
            "joinNumber": 0,
            "reserveRatio": null,
            "saleRate": null,
            "salePossibleYn": false,
            "lendingPossibleYn": false,
            "adult": false,
            "korGradeList": [{"code": "1001001", "name": "전체 관람가"}],
            "lastKoreanGrade": {"code": "1001001", "name": "전체 관람가"},
            "usGradeList": [],
            "lastUsGrade": null,
            "theater": 110,
            "formattedWeekendAudience": "0",
            "point": "0.00",
            "formattedReserveRatio": "0.00",
            "formattedSaleRate": "0.00"
        }, {
            "movieCode": 216214,
            "movieTitle": "시간을 꿈꾸는 소녀",
            "posterImageUrl": "/20221227_79/1672105992961sDeVc_JPEG/movie_image.jpg",
            "weekendAudience": 0,
            "openDate": "1.11",
            "pointSum": 0,
            "joinNumber": 0,
            "reserveRatio": null,
            "saleRate": null,
            "salePossibleYn": false,
            "lendingPossibleYn": false,
            "adult": false,
            "korGradeList": [{"code": "1001002", "name": "12세 관람가"}],
            "lastKoreanGrade": {"code": "1001002", "name": "12세 관람가"},
            "usGradeList": [],
            "lastUsGrade": null,
            "theater": 47,
            "formattedWeekendAudience": "0",
            "point": "0.00",
            "formattedReserveRatio": "0.00",
            "formattedSaleRate": "0.00"
        }, {
            "movieCode": 217479,
            "movieTitle": "해시태그 시그네",
            "posterImageUrl": "/20221214_128/1670995161996616NQ_JPEG/movie_image.jpg",
            "weekendAudience": 0,
            "openDate": "1.11",
            "pointSum": 0,
            "joinNumber": 0,
            "reserveRatio": null,
            "saleRate": null,
            "salePossibleYn": false,
            "lendingPossibleYn": false,
            "adult": false,
            "korGradeList": [{"code": "1001003", "name": "15세 관람가"}],
            "lastKoreanGrade": {"code": "1001003", "name": "15세 관람가"},
            "usGradeList": [],
            "lastUsGrade": null,
            "theater": 38,
            "formattedWeekendAudience": "0",
            "point": "0.00",
            "formattedReserveRatio": "0.00",
            "formattedSaleRate": "0.00"
        }, {
            "movieCode": 224601,
            "movieTitle": "10일간의 애인",
            "posterImageUrl": "/20230102_95/1672648466072TSUj0_JPEG/movie_image.jpg",
            "weekendAudience": 0,
            "openDate": "1.12",
            "pointSum": 0,
            "joinNumber": 0,
            "reserveRatio": null,
            "saleRate": null,
            "salePossibleYn": false,
            "lendingPossibleYn": false,
            "adult": false,
            "korGradeList": [{"code": "1001004", "name": "청소년 관람불가"}],
            "lastKoreanGrade": {"code": "1001004", "name": "청소년 관람불가"},
            "usGradeList": [],
            "lastUsGrade": null,
            "theater": 36,
            "formattedWeekendAudience": "0",
            "point": "0.00",
            "formattedReserveRatio": "0.00",
            "formattedSaleRate": "0.00"
        }, {
            "movieCode": 224922,
            "movieTitle": "문맨",
            "posterImageUrl": "/20221230_287/1672364963118GVQ50_JPEG/movie_image.jpg",
            "weekendAudience": 0,
            "openDate": "1.11",
            "pointSum": 0,
            "joinNumber": 0,
            "reserveRatio": null,
            "saleRate": null,
            "salePossibleYn": false,
            "lendingPossibleYn": false,
            "adult": false,
            "korGradeList": [{"code": "1001002", "name": "12세 관람가"}],
            "lastKoreanGrade": {"code": "1001002", "name": "12세 관람가"},
            "usGradeList": [],
            "lastUsGrade": null,
            "theater": 35,
            "formattedWeekendAudience": "0",
            "point": "0.00",
            "formattedReserveRatio": "0.00",
            "formattedSaleRate": "0.00"
        }, {
            "movieCode": 216173,
            "movieTitle": "나는 마을 방과후 교사입니다",
            "posterImageUrl": "/20221229_187/1672280630443nYnNs_JPEG/movie_image.jpg",
            "weekendAudience": 0,
            "openDate": "1.11",
            "pointSum": 0,
            "joinNumber": 0,
            "reserveRatio": null,
            "saleRate": null,
            "salePossibleYn": false,
            "lendingPossibleYn": false,
            "adult": false,
            "korGradeList": [{"code": "1001001", "name": "전체 관람가"}],
            "lastKoreanGrade": {"code": "1001001", "name": "전체 관람가"},
            "usGradeList": [],
            "lastUsGrade": null,
            "theater": 18,
            "formattedWeekendAudience": "0",
            "point": "0.00",
            "formattedReserveRatio": "0.00",
            "formattedSaleRate": "0.00"
        }, {
            "movieCode": 220792,
            "movieTitle": "엄마의 땅: 그리샤와 숲의 주인",
            "posterImageUrl": "/20221229_243/1672305832423TxG6x_JPEG/movie_image.jpg",
            "weekendAudience": 0,
            "openDate": "1.25",
            "pointSum": 0,
            "joinNumber": 0,
            "reserveRatio": null,
            "saleRate": null,
            "salePossibleYn": false,
            "lendingPossibleYn": false,
            "adult": false,
            "korGradeList": [{"code": "1001001", "name": "전체 관람가"}],
            "lastKoreanGrade": {"code": "1001001", "name": "전체 관람가"},
            "usGradeList": [],
            "lastUsGrade": null,
            "theater": 5,
            "formattedWeekendAudience": "0",
            "point": "0.00",
            "formattedReserveRatio": "0.00",
            "formattedSaleRate": "0.00"
        }, {
            "movieCode": 216209,
            "movieTitle": "유랑의 달",
            "posterImageUrl": "/20230105_129/1672911176159J1vx8_JPEG/movie_image.jpg",
            "weekendAudience": 0,
            "openDate": "1.18",
            "pointSum": 0,
            "joinNumber": 0,
            "reserveRatio": null,
            "saleRate": null,
            "salePossibleYn": false,
            "lendingPossibleYn": false,
            "adult": false,
            "korGradeList": [{"code": "1001003", "name": "15세 관람가"}],
            "lastKoreanGrade": {"code": "1001003", "name": "15세 관람가"},
            "usGradeList": [],
            "lastUsGrade": null,
            "theater": 5,
            "formattedWeekendAudience": "0",
            "point": "0.00",
            "formattedReserveRatio": "0.00",
            "formattedSaleRate": "0.00"
        }]
    }
} as any;