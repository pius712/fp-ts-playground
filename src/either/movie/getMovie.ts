import {pipe} from "fp-ts/function";
import * as E from "fp-ts/Either";
import {MovieResponse} from "./types";
import {pick} from "./pick";

export function getBoxOffice(response: MovieResponse) {
    return pipe(
        E.of(response),
        E.chain(pick('BOXOFFICE'))
    )
}

export function getComming(response: MovieResponse) {
    return pipe(
        E.of(response),
        E.chain(pick('COMMING'))
    )
}
