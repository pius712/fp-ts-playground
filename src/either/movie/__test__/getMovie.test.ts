import {boxOffice, comming} from "./fixture";
import {getBoxOffice, getComming} from "../getMovie";
import {getOrElse} from "fp-ts/Either";
import {pipe} from "fp-ts/function";
import {Movie} from "../types";

describe('getMoview', function () {

    it('get box office ', async () => {
        const boxoffice = pipe(getBoxOffice(boxOffice), getOrElse<Error, Movie[]>((e) => {
            throw e
        }))

        expect(boxoffice).toMatchSnapshot()
    })

    it('get box office2 ', async () => {
        const boxoffice = () => pipe(getBoxOffice(comming), getOrElse<Error, Movie[]>((e) => {
            throw e
        }))

        expect(boxoffice).toThrowErrorMatchingInlineSnapshot(`"해당 카테고리는 없습니다."`)


    })

})
