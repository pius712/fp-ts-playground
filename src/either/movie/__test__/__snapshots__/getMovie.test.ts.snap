// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getMoview get box office  1`] = `
[
  {
    "adult": false,
    "formattedReserveRatio": "0.00",
    "formattedSaleRate": "0.00",
    "formattedWeekendAudience": "1,274,184",
    "joinNumber": 0,
    "korGradeList": [
      {
        "code": "1001002",
        "name": "12세 관람가",
      },
    ],
    "lastKoreanGrade": {
      "code": "1001002",
      "name": "12세 관람가",
    },
    "lastUsGrade": null,
    "lendingPossibleYn": false,
    "movieCode": 74977,
    "movieTitle": "아바타: 물의 길",
    "openDate": null,
    "point": "0.00",
    "pointSum": 0,
    "posterImageUrl": "/20221215_185/1671091761840XXpCR_JPEG/movie_image.jpg",
    "reserveRatio": null,
    "salePossibleYn": false,
    "saleRate": null,
    "theater": 0,
    "usGradeList": [],
    "weekendAudience": 1274184,
  },
  {
    "adult": false,
    "formattedReserveRatio": "0.00",
    "formattedSaleRate": "0.00",
    "formattedWeekendAudience": "518,865",
    "joinNumber": 0,
    "korGradeList": [
      {
        "code": "1001002",
        "name": "12세 관람가",
      },
    ],
    "lastKoreanGrade": {
      "code": "1001002",
      "name": "12세 관람가",
    },
    "lastUsGrade": null,
    "lendingPossibleYn": false,
    "movieCode": 184509,
    "movieTitle": "영웅",
    "openDate": null,
    "point": "0.00",
    "pointSum": 0,
    "posterImageUrl": "/20221213_158/1670910727328mpqYu_JPEG/movie_image.jpg",
    "reserveRatio": null,
    "salePossibleYn": false,
    "saleRate": null,
    "theater": 0,
    "usGradeList": [],
    "weekendAudience": 518865,
  },
  {
    "adult": false,
    "formattedReserveRatio": "0.00",
    "formattedSaleRate": "0.00",
    "formattedWeekendAudience": "99,110",
    "joinNumber": 0,
    "korGradeList": [
      {
        "code": "1001003",
        "name": "15세 관람가",
      },
    ],
    "lastKoreanGrade": {
      "code": "1001003",
      "name": "15세 관람가",
    },
    "lastUsGrade": null,
    "lendingPossibleYn": false,
    "movieCode": 208508,
    "movieTitle": "젠틀맨",
    "openDate": null,
    "point": "0.00",
    "pointSum": 0,
    "posterImageUrl": "/20221228_75/1672203377094bQgf9_JPEG/movie_image.jpg",
    "reserveRatio": null,
    "salePossibleYn": false,
    "saleRate": null,
    "theater": 0,
    "usGradeList": [],
    "weekendAudience": 99110,
  },
  {
    "adult": false,
    "formattedReserveRatio": "0.00",
    "formattedSaleRate": "0.00",
    "formattedWeekendAudience": "91,483",
    "joinNumber": 0,
    "korGradeList": [
      {
        "code": "1001002",
        "name": "12세 관람가",
      },
    ],
    "lastKoreanGrade": {
      "code": "1001002",
      "name": "12세 관람가",
    },
    "lastUsGrade": null,
    "lendingPossibleYn": false,
    "movieCode": 220733,
    "movieTitle": "오늘 밤, 세계에서 이 사랑이 사라진다 해도",
    "openDate": null,
    "point": "0.00",
    "pointSum": 0,
    "posterImageUrl": "/20221104_258/1667527490266scSO3_JPEG/movie_image.jpg",
    "reserveRatio": null,
    "salePossibleYn": false,
    "saleRate": null,
    "theater": 0,
    "usGradeList": [],
    "weekendAudience": 91483,
  },
  {
    "adult": false,
    "formattedReserveRatio": "0.00",
    "formattedSaleRate": "0.00",
    "formattedWeekendAudience": "77,160",
    "joinNumber": 0,
    "korGradeList": [
      {
        "code": "1001003",
        "name": "15세 관람가",
      },
    ],
    "lastKoreanGrade": {
      "code": "1001003",
      "name": "15세 관람가",
    },
    "lastUsGrade": null,
    "lendingPossibleYn": false,
    "movieCode": 222301,
    "movieTitle": "올빼미",
    "openDate": null,
    "point": "0.00",
    "pointSum": 0,
    "posterImageUrl": "/20221123_65/1669168819647KdX9K_JPEG/movie_image.jpg",
    "reserveRatio": null,
    "salePossibleYn": false,
    "saleRate": null,
    "theater": 0,
    "usGradeList": [],
    "weekendAudience": 77160,
  },
  {
    "adult": false,
    "formattedReserveRatio": "0.00",
    "formattedSaleRate": "0.00",
    "formattedWeekendAudience": "47,672",
    "joinNumber": 0,
    "korGradeList": [
      {
        "code": "1001001",
        "name": "전체 관람가",
      },
    ],
    "lastKoreanGrade": {
      "code": "1001001",
      "name": "전체 관람가",
    },
    "lastUsGrade": null,
    "lendingPossibleYn": false,
    "movieCode": 223799,
    "movieTitle": "신비아파트 극장판 차원도깨비와 7개의 세계",
    "openDate": null,
    "point": "0.00",
    "pointSum": 0,
    "posterImageUrl": "/20221212_211/1670833558390SLzsN_JPEG/movie_image.jpg",
    "reserveRatio": null,
    "salePossibleYn": false,
    "saleRate": null,
    "theater": 0,
    "usGradeList": [],
    "weekendAudience": 47672,
  },
  {
    "adult": false,
    "formattedReserveRatio": "0.00",
    "formattedSaleRate": "0.00",
    "formattedWeekendAudience": "22,991",
    "joinNumber": 0,
    "korGradeList": [
      {
        "code": "1001001",
        "name": "전체 관람가",
      },
    ],
    "lastKoreanGrade": {
      "code": "1001001",
      "name": "전체 관람가",
    },
    "lastUsGrade": null,
    "lendingPossibleYn": false,
    "movieCode": 224788,
    "movieTitle": "눈의 여왕5: 스노우 프린세스와 미러랜드의 비밀",
    "openDate": null,
    "point": "0.00",
    "pointSum": 0,
    "posterImageUrl": "/20221221_206/16715988070200qULF_JPEG/movie_image.jpg",
    "reserveRatio": null,
    "salePossibleYn": false,
    "saleRate": null,
    "theater": 0,
    "usGradeList": [],
    "weekendAudience": 22991,
  },
  {
    "adult": false,
    "formattedReserveRatio": "0.00",
    "formattedSaleRate": "0.00",
    "formattedWeekendAudience": "19,315",
    "joinNumber": 0,
    "korGradeList": [
      {
        "code": "1001002",
        "name": "12세 관람가",
      },
    ],
    "lastKoreanGrade": {
      "code": "1001002",
      "name": "12세 관람가",
    },
    "lastUsGrade": null,
    "lendingPossibleYn": false,
    "movieCode": 201642,
    "movieTitle": "스위치",
    "openDate": null,
    "point": "0.00",
    "pointSum": 0,
    "posterImageUrl": "/20230104_71/1672799791128mA8Km_JPEG/movie_image.jpg",
    "reserveRatio": null,
    "salePossibleYn": false,
    "saleRate": null,
    "theater": 0,
    "usGradeList": [],
    "weekendAudience": 19315,
  },
  {
    "adult": false,
    "formattedReserveRatio": "0.00",
    "formattedSaleRate": "0.00",
    "formattedWeekendAudience": "15,362",
    "joinNumber": 0,
    "korGradeList": [
      {
        "code": "1001001",
        "name": "전체 관람가",
      },
    ],
    "lastKoreanGrade": {
      "code": "1001001",
      "name": "전체 관람가",
    },
    "lastUsGrade": null,
    "lendingPossibleYn": false,
    "movieCode": 222337,
    "movieTitle": "몬스터 신부: 101번째 프로포즈",
    "openDate": null,
    "point": "0.00",
    "pointSum": 0,
    "posterImageUrl": "/20221208_79/1670484193384Wix0b_JPEG/movie_image.jpg",
    "reserveRatio": null,
    "salePossibleYn": false,
    "saleRate": null,
    "theater": 0,
    "usGradeList": [],
    "weekendAudience": 15362,
  },
  {
    "adult": false,
    "formattedReserveRatio": "0.00",
    "formattedSaleRate": "0.00",
    "formattedWeekendAudience": "11,967",
    "joinNumber": 0,
    "lastUsGrade": null,
    "lendingPossibleYn": false,
    "movieCode": 222543,
    "movieTitle": "극장판 뽀로로와 친구들: 바이러스를 없애줘!",
    "openDate": null,
    "point": "0.00",
    "pointSum": 0,
    "posterImageUrl": "/20221108_225/1667872338018ez9cs_JPEG/movie_image.jpg",
    "reserveRatio": null,
    "salePossibleYn": false,
    "saleRate": null,
    "theater": 0,
    "usGradeList": [],
    "weekendAudience": 11967,
  },
]
`;
